### ISSUE TITLE
Analytics on the Website?

### ISSUE THREAD

**Issue by ryanleesipes** [thundernest/thunderbird-website#26]
Anyone really opposed to this? It would be nice to get some data on how much the website is being visited and by whom.

**Comment by wsmwk** [comment]
I should think it a priority, so we can guage the effectiveness of our messaging and marketing efforts.

**Comment by benbucksch** [comment]
Analytics, yes. Google Analytics, no. Google is a direct competitor.

Are there other analytics tools that can run on our servers and are sufficient for our needs? sancus, do you know of any?

**Comment by Sancus** [comment]
I've been meaning to add GA, yea. Mozilla uses Google Analytics extensively, so I can't imagine privacy or competitor concerns are all that significant.

Regardless, I'm open to alternatives, but they can't be log analysis - we don't have access to the logs because cloudflare serves most of our page views, not our actual servers, and they don't provide direct log access short of an Enterprise account that costs many thousands per month.

So it has to be some sort of in-page JS solution like GA.

**Comment by kewisch** [comment]
Most alternatives are not up to par or end up having other drawbacks. I am in favor of GA as long as we can use the Mozilla GA account.

**Comment by Sancus** [comment]
I don't know if we can use the Mozilla GA account. I would have to ask. I don't have access to add sites or users or anything like that, I can only see mozilla.org.

**Comment by mkmelin** [comment]
Why use the Mozilla GA account, instead of an own one? Isn't that just going to cause troublesome access for no apparent benefit?

**Comment by Sancus** [comment]
It's because Mozilla has a Google Analytics premium account which costs 150k/year(last I checked) and allows you to restrict Google from sharing your data with any other Google service or using it for adwords purposes. (See: [Bug 697436](https://bugzilla.mozilla.org/show_bug.cgi?id=697436#c14))

**Comment by mkmelin** [comment]
Right. But the comment you link to points to that that option is now available even for non-premium accounts as well.

**Comment by Sancus** [comment]
That's true, I could have sworn it said only other premium accounts have the options. Guess I was really tired when I read that, my bad. I checked the free GA signup and it has different Data Sharing options from that bug, which was 5 years ago. The new options sound somewhat more permissive to me but I'm not sure if GA Premium has the same ones or not.

**Comment by ryanleesipes** [comment]
I'm for Piwik, now [Matomo](https://matomo.org/) or GA. But we should just come to a decision on this so we can start getting it implemented.

**Comment by benbucksch** [comment]
Matomo +1

**Comment by Sancus** [comment]
We could use Matomo. It would cost ~$250/mo, since we get about 1.5M page views per month, for the cloud option. Of course we could host it ourselves but given that it requires a database server and web server, it's unlikely that that would be any cheaper. Plus it would take some amount of time to set up. 

Using Matomo does maybe make it possible to add on the start page since if we self-hosted we could conceivably do this without upsetting people by making requests to non-thunderbird servers at startup. So that might add interesting data. We would need a fairly robust(and also expensive) setup to do this though as I'm sure Matomo needs a pretty serious web server and database to accept 10M tracking pings per day. The cloud package would be something like $7000/mo for that much traffic.

I've also sent an email to the analytics team within Mozilla to ask them their opinion of data privacy and usage on free Google Analytics accounts given the current data sharing options available there as well.

We don't really need analytics on the site until mozilla.org redirects to it, but that is going to happen soon(in a couple of weeks, I would guess) so we want to have a solution in place before then.

**Comment by ryanleesipes** [comment]
Couldn't agree more @Sancus. @kewisch any thoughts? Let's see what Mozilla says, and then go from there. But I want to get this squared ASAP.

**Comment by kewisch** [comment]
I’m not really an analytics expert, but what I have heard from people who have spent a greater amount of time with the topic is that GA provides the best feature set, and the other solutions are not up to par. I would suggest to see what moz has to say about the privacy bit and if this is guaranteed then use that.

I don’t think we should be spending 7k per month on this if there is not a substantial benefit.

**Comment by benbucksch** [comment]
Food for thought regarding price: Why does Google offer it for [free](http://thednetworks.com/2011/10/16/you-are-not-the-customer-you-are-the-product-especially-when-it-comes-to-facebook/ )?

FWIW, I was suggesting to run Motomo on our servers. Obviously, that would be nuts to spend 7000 USD/month. Of course, our servers wouldn't be free either, but a dedicated 4-core with a 1 GB/s connection costs 50 EUR/month including traffic. That's what I would do.

Anyway, given that Ryan is responsible for the website, I defer the decision to him.

**Comment by ryanleesipes** [comment]
I'm taking this off this thread and into an Email to get some final details.

**Comment by Sancus** [comment]
The reply from Mozilla was basically that free, public GA has the same data sharing options as the Mozilla account does. Personally I would setup GA for the website only for now just because that option uses the least amount of my time.

I don't really know who has final say over this or what criteria we're using to decide, to be honest, but given that doing our own Matomo setup would use up a significant amount of my time(probably 10-20 hours tbh) I suspect it's Philipp's call. Doing a cloud Matomo setup for just the website would cost ~$250/mo and shouldn't take much more of my time than GA would, but will require rkent's involvement to enter payment information.

**Comment by benbucksch** [comment]
> our own Matomo setup ... probably 10-20 hours

+1 Personally, I think that's well worth it. Plus 200-300 Eur/month for 4-6 dedicated servers.

We'd also get data from users with ad blockers, which nowadays is around 25% of Firefox users. I am presuming that GA is blocked by them.

**Comment by MagicFab** [comment]
+1 for Matomo. I can help with setup, support and any questions about it. I can also help with updates and general administration of it. I am also MagicFab on mozillians.org.

**Comment by benbucksch** [comment]
MagicFab: Thanks! When you say "help with", do you mean that you can answer questions or that you can actually do the entire setup?

**Comment by mattab** [comment]
Hi guys, I'm the founder of Matomo and also a keen user of Thunderbird. We can directly offer 30% off the cloud service pricing of that helps? To get started you can get a free trial here https://www.innocraft.cloud/ and if you want to proceed please contact us for the discount. Btw we offer many more features than Google analytics and ability to move later from cloud to self hosted

**Comment by benbucksch** [comment]
mattab: We'd ideally also like to use it for the start page, then we'd have some 10-20 million pings per day. Do you know how many dedicated servers we'd need, if we were to self-host it?

**Comment by MagicFab** [comment]
@benbucksch both, although I wouldn't commit to the full initial setup by myself. I was part of the enterprise support team for PiwikPro a couple of years ago (up until just before v 3.0 release), so I could certainly provide guidance and time. 
I have mostly managed single-server setups for small projects and providers, so I'll let @mattab reply to your question about requirements but it's certainly multiple servers and several TB of RAID storage depending on how long you want to keep historical data.

**Comment by mattab** [comment]
Using on the start page is indeed a bit more involved. But we can help with providing you some info on how to manage your high traffic Matomo instance, how many servers to use, whether you need HA or a more simple setup. Could you please get in touch with us at contact@innocraft.com and we'll follow up and help out?

**Comment by mkmelin** [comment]
Not that Mamoto sounds pretty cool, but nevertheless I don't see the added value worth spending 4 figures monthly for it.
You have to be able to give a good answer to "if GA is good enough for Mozilla, why isn't it good enough for Thunderbird".

**Comment by benbucksch** [comment]
> I don't see the added value worth spending 4 figures monthly for it.

Agreed. Nobody is proposing that we choose the 7000 $/month plan. A self-hosted solution would cost ballpark of 300 Eur/month plus the setup. sancus said "probably 10-20 hours", maybe less given that we would have expert help (see above).

**Comment by Sancus** [comment]
OK, a lot of stuff is getting muddled here.

1) GA vs Matomo on the website E is $0 vs $175/mo(with the 30% discount so generously offered above). That's not too unreasonable!

2)  Aside from well-known bias that "google is bad", I'm not really convinced about there being **that much** of a privacy issue with GA. Mozilla uses it, and with user IP anonymization enabled(which we would certainly use), it is even compliant with German privacy laws. 

3) The 4-figure numbers are for using Matomo on the Start Page. Historically, there has been a substantial amount of resistance to using GA on the start page. I put it on there for a couple of weeks last year and there were a number of complaints. Even though there may not be any genuine privacy issue with GA, it is understandable if users are unhappy with the Thunderbird client making requests to Google servers on startup. That of course applies to cloud Matomo setups as well, unless they allow a custom domain. 

4) The 10-20 hour estimate was for self-hosting setup on AWS, where all I really need to do is set up the Matomo software itself and add it to the website. Dedicated hardware involves a whole host of problems that are completely inappropriate for a team with 1 person doing devops. For starters, we are talking about custom monitoring and backup solutions, and if we need multiple database servers with replication etc(seems likely), that all needs to be manually set up as well. Not to mention a load balancer, and all the network and OS settings that are involved with this whole thing... just so we have analytics. I'm not really interested in re-litigating this question every single time "hosting something on a server" comes up, so the answer is "not now, not ever" to dedicated servers.

That said, I don't know that AWS hosting of Matomo would cost anywhere near $7000/mo(that was the undiscounted cloud price for 10M pings per day). It's really hard to say without knowing anything about its performance level.

TLDR; Yes, this whole issue is too long. Lets forget about the Start Page for now. @ryanleesipes @kewisch Lets try a Matomo cloud setup on the website to start with. I assume that you two and maybe a couple of other council members will be the primary users of it, so after a few months we can re-evaluate, see if we're happy with it, and possibly start a discussion about adding analytics to the Start Page, which is a more more complex problem, and probably should involve some community discussion about what people are comfortable with Thunderbird doing on startup.

**Comment by benbucksch** [comment]
> bias that "google is bad"

No, data that companies (incl. Google) can use is bad:
https://www.google.com/search?q=cambridge+analytica

If I understood the fine print correctly, even with the checkboxes, Google still has the right to use the data for their own company goals. Google is directly marketing Chrome, too. It's marketing.

> Lets forget about the Start Page for now

OK for me.

> Lets try a Matomo cloud setup on the website to start with

**Comment by ryanleesipes** [comment]
I agree with @Sancus - thread is too long. Not sure is the proper place for this part of the conversation. Sending an Email to discuss feasibility of the options available.
